cpp/include/cudf/join.hpp

```diff
@@ -327,5 +327,26 @@ std::unique_ptr<cudf::table> left_anti_join(
   std::vector<cudf::size_type> const& return_columns,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief  Performs a cross join on two tables (left, right)
+ *
+ * The cross join returns the cartesian product of rows from each table.
```

Comment:
I'm not able to discern what the algorithm does from this description. Maybe it is because I don't know what a `cross_join` is. And is this the right name for it? There are already a plethora of names in the space: 
* `inner_product` from C++, APL & Haskell
* `cross` from Q
* `cartesian_product` from D & the C++ ranges-v3 library
* `cross` or `cross_product` from other FP languages

Reply:
Cross-join is indeed the correct name for this. https://en.wikipedia.org/wiki/Join_(SQL)#Cross_join

The confusion arises from cudf borrowing semantics from C++, pandas, and SQL. This one comes from SQL.

Reply:
I think a pseudo-code example should help illuminate things.